tensorboard/components/tf_ng_tensorboard/header/containers/header.component.test.ts

```diff
@@ -145,4 +150,24 @@ describe('header.component', () => {
     expect(dispatch).toHaveBeenCalledTimes(1);
     expect(dispatch).toHaveBeenCalledWith(changePlugin({plugin: 'bar'}));
   });
+
+  it('does not keep observer after getting the most recent value', async () => {
```

Comment:
This test pattern seems to mix presentation-level concerns (e.g., `By.css(...)` and `click()`) with data model/connector concerns, i.e. ngrx/rxjs stuff.

Reply:
Sorry, what are you suggesting here? Are you suggesting that I should stub the callback of the click?

Reply:
Similar to suggestions elsewhere re separating the store from the view.  You could have separate tests for 1) a click fires the appropriate action, 2) upon receiving the action, the store updates appropriately, and 3) there are no memory leaks in the form of observers that should have been unsubscribed.

Per the other comment I'm still unclear on what this is meant to test.  The title suggests #3, in which case I'd expect you can test that more directly, without involving a click.